lib/rpm.rb

```diff
@@ -41,10 +41,22 @@ def self.transaction(root = '/')
   # @param [String] name Name of the macro
   # @return [String] value of macro +name+
   def self.[](name)
-    buffer = ::FFI::MemoryPointer.new(:pointer, 1024)
-    buffer.write_string("%{#{name}}")
-    ret = RPM::C.expandMacros(nil, nil, buffer, 1024)
-    buffer.read_string
+    if C::rpm_version_code >= ((4 << 16) + (14 << 8) + (0 << 0))
+      obuf = ::FFI::MemoryPointer.new(:pointer)
+      sbuf = FFI::MemoryPointer.from_string("%{#{name}}")
+      ret = RPM::C.rpmExpandMacros(nil, sbuf, obuf, 0)
+      return if ret < 0
```

Comment:
Should we just return or should we raise an exception?  From a caller perspective, this method is called like Rpm['foo'] so I think the caller expects it to expand...if it doesn't it will return nil, which feels wrong?

Reply:
Yeah that's true I wonder if the return value has any pre-defined error codes that can help raise a useful exception

Reply:
I'm ok with merging as is to get things green, and we can research it more later - it's up to you

Reply:
I'll just raise a normal RuntimeError for now